sbin/shipper_utils/systemctl_wrapper.py

```diff
@@ -0,0 +1,210 @@
+    print_status_message(status, mode, daemon, target)
+    verify_systemctl_status_code(status, mode, daemon, target, disable_on_failure=True)
+    greatest_status = max(greatest_status, status)
+  return greatest_status
+
+# This function performs a systemctl command (mode) on a given daemon on a given machine
+def perform_systemctl_command_on_worker(daemon, mode, target):
+  if not target in WORKERS:
+    print("There is no machine with the key {0}".format(target))
+    sys.exit(EXIT_CODES['bad_arguments'])
+
+  user = WORKERS[target]['username']
+  host = WORKERS[target]['address']
+
+  if host == 'localhost' or user == '':
+    print("Please don't specify machine id if you wish to run locally.")
+    sys.exit(EXIT_CODES['bad_arguments'])
+
+  script_directory = os.path.join(INSTALL_DIR, 'sbin', 'shipper_utils', 'systemctl_wrapper.py')
+  command = "sudo {0} {1} --daemon {2}".format(script_directory, mode, daemon)
+  try:
+      ssh = paramiko.SSHClient()
+      ssh.get_host_keys()
+      ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+      ssh.connect(hostname = host, username = user)
+  except Exception as e:
+      print("ERROR: could not ssh to {0}@{1} due to following error: {2}".format(user, host,str(e)))
+      return EXIT_CODES['failure']
+  try:
+      (stdin, stdout, stderr) = ssh.exec_command(command)
+      status = int(stdout.channel.recv_exit_status())
+  except Exception as e:
+      print("ERROR: Command did not properly execute: ".format(host, str(e)))
+      status = EXIT_CODES['failure']
+  finally:
+      ssh.close()
+      return status
+
+def disable_machine(target):
+  if WORKERS == None:
+    print('Cannot disable as autograding_workers.json does not exist.')
+    return
+  print('Disabling', target)
+  WORKERS[target]['enabled'] = False
+
+  with open(AUTOGRADING_WORKERS_PATH, 'w') as json_file:
+    json.dump(WORKERS, json_file, indent=4)
+
+def verify_systemctl_status_code(status, mode, daemon, target, disable_on_failure=False):
+  if not mode in VALID_MODES:
+    print("ERROR: Invalid mode")
+    return False
+
+  correct = True
+  if (mode == 'start' or mode == 'restart') and status != EXIT_CODES['active']:
+    correct = False
+  elif mode == 'stop' and status != EXIT_CODES['inactive']:
+    correct = False
+
+  if correct == False:
+    print("ERROR: Could not {0} the {1} daemon on {2}".format(mode, daemon, target))
+    if disable_on_failure:
+      disable_machine(target)
+
+  return correct
+
+if __name__ == "__main__":
+  #parse arguments
+  parser = argparse.ArgumentParser(description='A wrapper for the various systemctl functions. \
+    This script must be run as the submitty supervisor.',
+    epilog='ERROR CODES: 0 = After performing the specified action, the requested daemon is dead, 1 = After performing the \
+    specified action, the requested daemon is alive, 2 = Failure performing operation, 3 = invalid mode, daemon, or target')
```

Comment:
For readability, we probably want to put in newlines and do something like: 
```
epilog="""ERROR CODES:
0 = After performing the specified action, the requested daemon is dead
1 = After performing the specified action, the requested daemon is alive
2 = Failure performing operation
3 = invalid mode, daemon, or target"""
```

Reply:
I will do this.